src/query/query_parse.rs

```diff
@@ -0,0 +1,131 @@
+use crate::query::query_types::*;
+use pest::Parser;
+use crate::query::QueryError;
+
+#[derive(Parser)]
+#[grammar = "query/query.pest"]
+struct QueryParser;
+
+pub fn parse_query(query: &str) -> Result<QueryNode, QueryError> {
+    match QueryParser::parse(Rule::query, query) {
+        Ok(pairs) => {
+            // Below never fails as query has been parsed and we know query rule can only have 1
+            // expr and its inner can only be length 1 also (either latest or a string)
+            let query = pairs.peek().unwrap().into_inner().peek().unwrap();
+            parse_query_content(query)
+        }
+        Err(e) => Err(QueryError::ParseError(Box::new(e))),
+    }
+}
+
+fn parse_query_content(query: pest::iterators::Pair<Rule>) -> Result<QueryNode, QueryError> {
+    match query.as_rule() {
+        Rule::string => {
+            let x = get_string_inner(query);
+            Ok(QueryNode::Lookup(LookupLhs::Id, x))
+        }
+        Rule::noVariableFunc => {
+            // Pest asserts for us that the only possible no variable func is latest()
+            // we might want to move this validation into Rust code later to return
+            // better errors to the user
+            Ok(QueryNode::Latest(None))
+        }
+        Rule::infixExpression => {
+            let mut infix = query.into_inner();
+            let first_arg = infix.next().unwrap();
+            let infix_function = infix.next().unwrap();
+            let second_arg = infix.next().unwrap();
+            match infix_function.as_str() {
+                "==" => {
+                    let lookup_type = match first_arg.as_str() {
+                        "id" => LookupLhs::Id,
+                        "name" => LookupLhs::Name,
+                        _ => unreachable!(),
+                    };
+                    let search_term = get_string_inner(second_arg.into_inner().peek().unwrap());
+                    Ok(QueryNode::Lookup(lookup_type, search_term))
+                }
+                _ => {
+                    let err = pest::error::Error::new_from_span(
+                        pest::error::ErrorVariant::CustomError {
+                            message: format!(
+                                "Encountered unknown infix operator: {}",
```

Comment:
```
cargo run --bin outpacl_query -- -r tests/example/ -q 'name != 123'
```

does not actually throw this - possibly the parser is swallowing it. Code coverage would make this easier to test

Reply:
Thanks, fixed this now. You should now get

```
cargo run --bin outpack_query -- -r tests/example/ -q "name != \"123\""
```

```
Failed to parse query
 --> 1:6
  |
1 | name != "123"
  |      ^^
  |
  = Encountered unknown infix operator: !=

Process finished with exit code 1

```